pkg/controller/provider/web/vsphere/client.go

```diff
@@ -82,3 +89,131 @@ func (r *Resolver) Path(resource interface{}, id string) (path string, err error
 
 	return
 }
+
+//
+// Resource finder.
+type Finder struct {
+}
+
+//
+// Find a resource by ref.
+func (r *Finder) ByRef(client base.Client, resource interface{}, ref base.Ref) (err error) {
```

Comment:
This is the _meat_ of the find-by-ref capability.
Performs a `List()` and if the query matches multiple resources (list contains more than one) raises and `ambiguous` error.  Else, updates the object passed in with list[0].